docs/modules/ROOT/examples/io/quarkiverse/langchain4j/samples/CustomProvider.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkiverse.langchain4j.samples;
+
+import java.util.function.Supplier;
+
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.store.memory.chat.InMemoryChatMemoryStore;
+
+public class CustomProvider implements Supplier<ChatMemoryProvider> {
+    @Override
+    public ChatMemoryProvider get() {
+        return new ChatMemoryProvider() {
+
+            @Override
+            public ChatMemory get(Object memoryId) {
+                return MessageWindowChatMemory.builder()
+                        .maxMessages(20)
+                        .id(memoryId)
+                        .chatMemoryStore(new InMemoryChatMemoryStore())
```

Comment:
Does it make sense to create a new store here every time?

Reply:
Good catch! I fixed it